corehq/apps/locations/views.py

```diff
@@ -834,6 +834,16 @@ def location_export(request, domain):
     return response
 
 
+def get_locations_paylod_from_ids(ids, domain, base_queryset):
+    from corehq.apps.locations.util import get_locations_from_ids
+    locations = get_locations_from_ids(ids, domain, base_queryset=base_queryset)
+    return [loc_to_payload(loc) for loc in locations]
+
+
+def base_queryset_for_domain_and_user(domain, user):
+    return SQLLocation.objects.accessible_to_user(domain, user)
```

Comment:
I don't think we need to worry about admin users, because if code is calling `accessible_to_user(domain, user)`, then they are specifying a domain anyways.  The only time I can think of when we might be intentionally dealing with a cross-domain queryset is in migrations and very advanced type things which presumably wouldn't be using `accessible_to_user` anyways.

One other point - you wrote:
> to be
```python
if user.has_permission(domain, 'access_all_locations'):
    return SQLLocation.active_objects.filter(domain=domain)
```

But it should actually be:
```python
if user.has_permission(domain, 'access_all_locations'):
    return self.filter(domain=domain)
```
This way it will chain properly with other filters that had been applied.